### PR TITLE
add keeps (libraries only) on kifi.com [FE]

### PR DESCRIPTION
**Pull request by ahsu1230**
@andrewconner @yipingliao 

numKeeps per library on left panel increases with addKeepsToLibrary()
may still need to work on several other bugs
(keeping urls in different libraries, unkeeping from Library, etc.)

### CODE REVIEW

File: kifi-backend/modules/shoebox/angular/src/keep/addKeep.tpl.html [hunk 1]
```diff
@@ -7,11 +7,19 @@
...
+                    value="{{lib.id}}">{{lib.name}}</option>
+          </select>
+        </div>
       </div>
     </div>
   </div>
   <div class="dialog-buttons">
-    <button class="kf-action-button" ng-click="keepUrl()" tabindex="0">Keep it</button>
+    <button class="kf-action-button" ng-if="!librariesEnabled" ng-click="keepUrl()" tabindex="0">Keep it</button>
+    <button class="kf-action-button" ng-if="librariesEnabled" ng-click="keepToLibrary()" tabindex="0">Keep To Library</button>
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/keep/addKeep.tpl.html]: I noticed that this text doesn't show up entirely on the button. Maybe we need to style the button to be longer to accommodate for this?

**Comment by yipingliao** [comment]
Looks good!

File: kifi-backend/modules/shoebox/angular/src/keep/addKeep.js [hunk 2]
```diff
@@ -94,6 +94,32 @@ angular.module('kifi')
...
+          } else {
+            scope.state.invalidUrl = true;
+          }
+        };
+
+        scope.libraries = _.filter(libraryService.librarySummaries, function(lib) {
+          return lib.access !== "read_only";
+        });
+        scope.data = scope.data || {};
+        scope.data.selectedLibraryId = scope.libraries[0].id;
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/src/keep/addKeep.js]: May be nice to look for the Main (look up `_.find`)

File: kifi-backend/modules/shoebox/angular/src/keeps/keepService.js [hunk 3]
```diff
@@ -473,6 +473,33 @@ angular.module('kifi')
         return processKeepAction(data);
       },
 
+      keepToLibrary: function (keepUrls, libraryId) {
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/src/keeps/keepService.js:476]: @yipingliao Is keepService the place to do this? Up to you, I know you're moving networking aspects of keepService around?

File: kifi-backend/modules/shoebox/angular/src/keep/addKeep.tpl.html (hunk 1 above)
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/angular/src/keep/addKeep.tpl.html]: We could also leave it as just "Keep It"
I don't think we need to explicitly say Keep To Library - I had that just to make sure I was in the library experiment.

**Comment by andrewconner** [comment]
Feel free to merge when ready.

File: kifi-backend/modules/shoebox/angular/src/keep/addKeep.js [hunk 4]
```diff
@@ -94,6 +94,38 @@ angular.module('kifi')
           }
         };
 
+        scope.keepToLibrary = function () {
+          var url = (scope.state.input) || '';
+          if (url && keepService.validateUrl(url)) {
+            var libUrl = _.find(scope.libraries, function(lib) {
+                return lib.id === scope.data.selectedLibraryId
+              }).url;
+            $location.path(libUrl);
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/keep/addKeep.js]: Maybe not re-route? Stay on currently working view.

File: kifi-backend/modules/shoebox/angular/src/keeps/keepService.js (hunk 3 above)
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/keeps/keepService.js:476]: Just discussed with @ahsu1230 . It's good there for now.

File: kifi-backend/modules/shoebox/angular/src/keep/addKeep.js (hunk 4 above)
Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/src/keep/addKeep.js]: Yeah, not sure. It was there before for one case... you think it's better to stay? I'm cool with that.

Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/src/keep/addKeep.js]: Also, if we do redirect, probably should _after_ adding it. And the keep specific redirect below should go.

**Pull request merged**